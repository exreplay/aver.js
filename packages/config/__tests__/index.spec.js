import path from 'path';
import { getAverjsConfig } from '../index';

beforeEach(() => {
  process.env.PROJECT_PATH = path.resolve(__dirname, '../__fixtures__/aver-config.js');
});

test('should change progressbar option to false', () => {
  const config = getAverjsConfig();
  expect(config.progressbar).toBe(false);
});

test('should change nested attribute extract to true', () => {
  const config = getAverjsConfig();
  expect(config.webpack.css.extract).toBe(true);
});

test('should return a default config when no config file is present', () => {
  process.env.PROJECT_PATH = '';
  const config = getAverjsConfig();
  expect(config.progressbar).toBe(true);
});
