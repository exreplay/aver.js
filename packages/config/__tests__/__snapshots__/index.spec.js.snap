// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match default snapshot config when no config file is present 1`] = `
Object {
  "aliases": Object {},
  "createRenderer": Object {},
  "csrf": true,
  "csrfExclude": Array [],
  "i18n": Object {
    "silentTranslationWarn": true,
  },
  "plugins": Array [],
  "progressbar": true,
  "router": Object {},
  "webpack": Object {
    "base": false,
    "client": false,
    "css": Object {
      "extract": false,
      "styleResources": Object {
        "options": Object {},
        "resources": Array [],
      },
    },
    "process": Object {
      "env": Object {},
    },
    "purgeCss": false,
    "server": false,
    "sw": false,
  },
}
`;
